pkg/operator/ceph/file/controller.go

```diff
@@ -74,6 +73,12 @@ type ReconcileCephFilesystem struct {
 	context         *clusterd.Context
 	cephClusterSpec *cephv1.ClusterSpec
 	clusterInfo     *cephclient.ClusterInfo
+	fsChannels      map[string]*fsHealth
```

Comment:
Why do we need one channel per FS? We could keep it simple with a single goroutine that iterates over all the filesystems at the check interval. The goroutine overhead isn't the concern, just trying to keep the code simpler.

Reply:
This is not how the health checkers are designed today. It's per `CephFilesystem` today. It could be refactored for sure, this would mean changing the pool/object and this implementation. It's not urgent IMO :)